# Flight Operations Data Analysis - Day 10

**Assignment:** Flight Operations Data Analysis (Day 10)
Create a Complete Data Analysis Notebook using the provided Flight Operations dataset (uploaded in 'Study Material' tab of LMS) and Pandas. Follow a complete data analysis workflow by loading and understanding the dataset, inspecting its structure and statistics, selecting and filtering relevant data, sorting records, performing grouping and aggregation, and applying suitable data transformations using the Pandas concepts covered throughout the week. Use appropriate Pandas operations to answer meaningful questions about the dataset, identify patterns and key findings, and write 5 to 8 observations based on your analysis. Present the entire process clearly in a well-organized Google Colab Notebook and upload it on GitHub before attaching the GitHub link here.



In [37]:
import pandas as pd

In [38]:
data = pd.read_csv("Day10_Flight_Operations_Dataset.csv")
data.head(10)

,Flight_ID,Flight_Date,Airline,Origin,Destination,Aircraft,Travel_Class,Passengers,Seat_Capacity,Average_Ticket_Price,Delay_Minutes,Flight_Status,Weather,Booking_Channel,Avg_Baggage_Kg,Meal_Preference,Passenger_Satisfaction
0,FL0001,2026-03-23,SpiceJet,Mumbai,Bengaluru,Airbus A319,Economy,210,180,6894,50.0,Delayed,Storm,Online Travel Portal,23,No Meal,3
1,FL0002,2026-06-07,Akasa Air,Delhi,Mumbai,Boeing 737,Economy,70,210,4468,0.0,On Time,Fog,Travel Agency,16,Vegan,5
2,FL0003,2026-05-12,Air India,Kochi,Delhi,Boeing 737,Economy,67,220,4713,5.0,On Time,Rain,Travel Agency,19,No Meal,2
3,FL0004,2026-05-30,Akasa Air,Srinagar,Delhi,Airbus A319,Economy,153,220,3791,35.0,Delayed,Storm,Travel Agency,18,No Meal,2
4,FL0005,2026-02-27,Air India,Pune,Delhi,Airbus A319,Premium Economy,178,180,6505,20.0,Delayed,Cloudy,Airline Website,25,No Meal,5
5,FL0006,2026-05-20,Akasa Air,Kochi,Delhi,Airbus A319,Economy,123,220,5636,35.0,Delayed,Cloudy,Travel Agency,13,Vegetarian,2
6,FL0007,2026-06-17,Air India Express,Delhi,Srinagar,Airbus A319,Premium Economy,178,210,9430,0.0,On Time,Storm,Online Travel Portal,18,Vegetarian,4
7,FL0008,2026-04-28,Air India,Delhi,Chennai,Airbus A319,Business,157,220,14173,0.0,On Time,Fog,Airline Website,25,Vegan,5
8,FL0009,2026-01-25,Akasa Air,Delhi,Chennai,Airbus A320,Economy,82,210,5893,75.0,Delayed,Cloudy,Mobile App,8,Vegetarian,4
9,FL0010,2026-06-12,Akasa Air,Ahmedabad,Bengaluru,Airbus A321,Economy,125,220,2500,12.0,On Time,Cloudy,Online Travel Portal,23,Non-Vegetarian,4


## 1. Inspecting Dataset Structure & Statistics

In [39]:
data.shape

(180, 17)

In [40]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 180 entries, 0 to 179
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Flight_ID               180 non-null    object 
 1   Flight_Date             180 non-null    object 
 2   Airline                 180 non-null    object 
 3   Origin                  180 non-null    object 
 4   Destination             180 non-null    object 
 5   Aircraft                180 non-null    object 
 6   Travel_Class            180 non-null    object 
 7   Passengers              180 non-null    int64  
 8   Seat_Capacity           180 non-null    int64  
 9   Average_Ticket_Price    180 non-null    int64  
 10  Delay_Minutes           173 non-null    float64
 11  Flight_Status           180 non-null    object 
 12  Weather                 180 non-null    object 
 13  Booking_Channel         180 non-null    object 
 14  Avg_Baggage_Kg          180 non-null    in

In [41]:
data.describe()

,Passengers,Seat_Capacity,Average_Ticket_Price,Delay_Minutes,Avg_Baggage_Kg,Passenger_Satisfaction
count,180.000000,180.000000,180.000000,173.000000,180.000000,180.000000
mean,122.772222,197.238889,6111.677778,19.335260,15.150000,3.494444
std,45.635651,15.408176,3615.220040,29.029712,5.705506,1.169908
min,45.000000,180.000000,1949.000000,0.000000,5.000000,2.000000
25%,78.750000,186.000000,3815.500000,0.000000,10.000000,2.000000
50%,125.000000,189.000000,5380.000000,5.000000,15.500000,3.000000
75%,162.250000,210.000000,6790.000000,20.000000,20.000000,5.000000
max,210.000000,220.000000,27832.000000,110.000000,25.000000,5.000000


In [42]:
data.columns.tolist()

['Flight_ID',
 'Flight_Date',
 'Airline',
 'Origin',
 'Destination',
 'Aircraft',
 'Travel_Class',
 'Passengers',
 'Seat_Capacity',
 'Average_Ticket_Price',
 'Delay_Minutes',
 'Flight_Status',
 'Weather',
 'Booking_Channel',
 'Avg_Baggage_Kg',
 'Meal_Preference',
 'Passenger_Satisfaction']

In [43]:
data.isnull().sum()

Flight_ID                 0
Flight_Date               0
Airline                   0
Origin                    0
Destination               0
Aircraft                  0
Travel_Class              0
Passengers                0
Seat_Capacity             0
Average_Ticket_Price      0
Delay_Minutes             7
Flight_Status             0
Weather                   0
Booking_Channel           0
Avg_Baggage_Kg            0
Meal_Preference           0
Passenger_Satisfaction    0
dtype: int64

## 2. Data Cleaning & Transformations

In [44]:
data["Delay_Minutes"] = data["Delay_Minutes"].fillna(
    data["Delay_Minutes"].median()
)

data["Delay_Minutes"].isnull().sum()

np.int64(0)

In [45]:
data["Flight_Date"] = pd.to_datetime(data["Flight_Date"])
data.head()

,Flight_ID,Flight_Date,Airline,Origin,Destination,Aircraft,Travel_Class,Passengers,Seat_Capacity,Average_Ticket_Price,Delay_Minutes,Flight_Status,Weather,Booking_Channel,Avg_Baggage_Kg,Meal_Preference,Passenger_Satisfaction
0,FL0001,2026-03-23,SpiceJet,Mumbai,Bengaluru,Airbus A319,Economy,210,180,6894,50.0,Delayed,Storm,Online Travel Portal,23,No Meal,3
1,FL0002,2026-06-07,Akasa Air,Delhi,Mumbai,Boeing 737,Economy,70,210,4468,0.0,On Time,Fog,Travel Agency,16,Vegan,5
2,FL0003,2026-05-12,Air India,Kochi,Delhi,Boeing 737,Economy,67,220,4713,5.0,On Time,Rain,Travel Agency,19,No Meal,2
3,FL0004,2026-05-30,Akasa Air,Srinagar,Delhi,Airbus A319,Economy,153,220,3791,35.0,Delayed,Storm,Travel Agency,18,No Meal,2
4,FL0005,2026-02-27,Air India,Pune,Delhi,Airbus A319,Premium Economy,178,180,6505,20.0,Delayed,Cloudy,Airline Website,25,No Meal,5


In [46]:
data = data.drop_duplicates()
data.shape

(180, 17)

In [47]:
data["Load_Factor"] = (data["Passengers"] / data["Seat_Capacity"]) * 100
data["Month"] = data["Flight_Date"].dt.month
data.head()

,Flight_ID,Flight_Date,Airline,Origin,Destination,Aircraft,Travel_Class,Passengers,Seat_Capacity,Average_Ticket_Price,Delay_Minutes,Flight_Status,Weather,Booking_Channel,Avg_Baggage_Kg,Meal_Preference,Passenger_Satisfaction,Load_Factor,Month
0,FL0001,2026-03-23,SpiceJet,Mumbai,Bengaluru,Airbus A319,Economy,210,180,6894,50.0,Delayed,Storm,Online Travel Portal,23,No Meal,3,116.666667,3
1,FL0002,2026-06-07,Akasa Air,Delhi,Mumbai,Boeing 737,Economy,70,210,4468,0.0,On Time,Fog,Travel Agency,16,Vegan,5,33.333333,6
2,FL0003,2026-05-12,Air India,Kochi,Delhi,Boeing 737,Economy,67,220,4713,5.0,On Time,Rain,Travel Agency,19,No Meal,2,30.454545,5
3,FL0004,2026-05-30,Akasa Air,Srinagar,Delhi,Airbus A319,Economy,153,220,3791,35.0,Delayed,Storm,Travel Agency,18,No Meal,2,69.545455,5
4,FL0005,2026-02-27,Air India,Pune,Delhi,Airbus A319,Premium Economy,178,180,6505,20.0,Delayed,Cloudy,Airline Website,25,No Meal,5,98.888889,2


## 3. Selecting & Filtering Relevant Data

In [48]:
pricy_tickets = data[
    (data["Average_Ticket_Price"] > 6000) &
    (
        data["Origin"].isin(["Bengaluru", "Mumbai", "Srinagar"]) |
        data["Destination"].isin(["Bengaluru", "Mumbai", "Srinagar"])
    ) &
    (data["Travel_Class"] == "Premium Economy")
]
pricy_tickets

,Flight_ID,Flight_Date,Airline,Origin,Destination,Aircraft,Travel_Class,Passengers,Seat_Capacity,Average_Ticket_Price,Delay_Minutes,Flight_Status,Weather,Booking_Channel,Avg_Baggage_Kg,Meal_Preference,Passenger_Satisfaction,Load_Factor,Month
6,FL0007,2026-06-17,Air India Express,Delhi,Srinagar,Airbus A319,Premium Economy,178,210,9430,0.0,On Time,Storm,Online Travel Portal,18,Vegetarian,4,84.761905,6
15,FL0016,2026-05-15,Akasa Air,Mumbai,Bengaluru,Boeing 737,Premium Economy,69,186,7860,0.0,On Time,Storm,Mobile App,5,Vegan,3,37.096774,5
20,FL0021,2026-06-23,Vistara,Srinagar,Delhi,Airbus A320,Premium Economy,173,189,9018,0.0,On Time,Fog,Airline Website,14,Vegan,4,91.534392,6
25,FL0026,2026-05-11,Air India,Ahmedabad,Bengaluru,Boeing 737,Premium Economy,137,186,8836,12.0,On Time,Fog,Online Travel Portal,22,No Meal,4,73.655914,5
28,FL0029,2026-02-06,Air India Express,Mumbai,Bengaluru,Airbus A321,Premium Economy,185,189,8098,0.0,On Time,Rain,Travel Agency,22,Vegan,4,97.883598,2
30,FL0031,2026-01-08,Vistara,Delhi,Srinagar,Airbus A321,Premium Economy,209,180,9962,50.0,Delayed,Cloudy,Mobile App,9,Vegetarian,4,116.111111,1
34,FL0035,2026-04-24,Vistara,Delhi,Srinagar,Boeing 737,Premium Economy,77,210,11441,0.0,On Time,Fog,Travel Agency,9,Vegan,3,36.666667,4
36,FL0037,2026-06-23,Vistara,Bengaluru,Kochi,Airbus A320,Premium Economy,184,220,11614,0.0,On Time,Fog,Mobile App,5,Vegan,4,83.636364,6
56,FL0057,2026-04-21,SpiceJet,Mumbai,Kolkata,Airbus A320,Premium Economy,51,189,10530,12.0,On Time,Rain,Mobile App,23,Vegetarian,3,26.984127,4
66,FL0067,2026-04-23,Vistara,Mumbai,Kolkata,Boeing 737,Premium Economy,109,210,8473,20.0,Delayed,Cloudy,Airline Website,24,Vegetarian,3,51.904762,4


In [51]:
customer_satisfication = data[data["Passenger_Satisfaction"] >= 4]
customer_satisfication.value_counts()

Flight_ID  Flight_Date  Airline            Origin     Destination  Aircraft     Travel_Class     Passengers  Seat_Capacity  Average_Ticket_Price  Delay_Minutes  Flight_Status  Weather  Booking_Channel       Avg_Baggage_Kg  Meal_Preference  Passenger_Satisfaction  Load_Factor  Month
FL0002     2026-06-07   Akasa Air          Delhi      Mumbai       Boeing 737   Economy          70          210            4468                  0.0            On Time        Fog      Travel Agency         16              Vegan            5                       33.333333    6        1
FL0005     2026-02-27   Air India          Pune       Delhi        Airbus A319  Premium Economy  178         180            6505                  20.0           Delayed        Cloudy   Airline Website       25              No Meal          5                       98.888889    2        1
FL0007     2026-06-17   Air India Express  Delhi      Srinagar     Airbus A319  Premium Economy  178         210            9430             

In [50]:
delay_flights = data[data["Delay_Minutes"] > 50]
delay_flights

,Flight_ID,Flight_Date,Airline,Origin,Destination,Aircraft,Travel_Class,Passengers,Seat_Capacity,Average_Ticket_Price,Delay_Minutes,Flight_Status,Weather,Booking_Channel,Avg_Baggage_Kg,Meal_Preference,Passenger_Satisfaction,Load_Factor,Month
8,FL0009,2026-01-25,Akasa Air,Delhi,Chennai,Airbus A320,Economy,82,210,5893,75.0,Delayed,Cloudy,Mobile App,8,Vegetarian,4,39.047619,1
21,FL0022,2026-02-01,Vistara,Kochi,Delhi,Airbus A321,Economy,126,220,5887,75.0,Delayed,Clear,Airline Website,10,Vegetarian,2,57.272727,2
26,FL0027,2026-02-10,SpiceJet,Bengaluru,Hyderabad,Airbus A319,Economy,169,180,3973,110.0,Delayed,Clear,Mobile App,7,Vegan,4,93.888889,2
29,FL0030,2026-04-10,SpiceJet,Delhi,Srinagar,Airbus A321,Economy,200,210,2716,75.0,Delayed,Storm,Airline Website,10,Vegan,3,95.238095,4
50,FL0051,2026-05-03,Air India,Delhi,Mumbai,Airbus A321,Economy,135,210,5129,110.0,Delayed,Rain,Online Travel Portal,17,Vegetarian,4,64.285714,5
57,FL0058,2026-04-14,IndiGo,Hyderabad,Kolkata,Boeing 737,Premium Economy,151,210,6834,110.0,Delayed,Cloudy,Travel Agency,16,Vegetarian,4,71.904762,4
82,FL0083,2026-04-18,Air India Express,Lucknow,Delhi,Airbus A319,Economy,71,180,2484,110.0,Delayed,Cloudy,Online Travel Portal,22,Vegetarian,4,39.444444,4
107,FL0108,2026-05-30,IndiGo,Lucknow,Delhi,Airbus A319,Premium Economy,198,180,10551,75.0,Delayed,Rain,Travel Agency,13,Vegan,2,110.000000,5
128,FL0129,2026-01-04,SpiceJet,Lucknow,Delhi,Boeing 737,Premium Economy,67,210,5657,110.0,Delayed,Cloudy,Online Travel Portal,21,Non-Vegetarian,2,31.904762,1
131,FL0132,2026-02-04,Air India,Hyderabad,Kolkata,Airbus A321,Economy,61,220,3483,75.0,Delayed,Clear,Mobile App,18,No Meal,5,27.727273,2


## 4. Sorting Records

In [52]:
data.sort_values(
    by="Average_Ticket_Price",
    ascending=False
)[[
    "Flight_ID",
    "Airline",
    "Origin",
    "Destination",
    "Travel_Class",
    "Average_Ticket_Price"
]].head(5)

,Flight_ID,Airline,Origin,Destination,Travel_Class,Average_Ticket_Price
138,FL0139,Air India Express,Delhi,Chennai,Business,27832
52,FL0053,Akasa Air,Mumbai,Bengaluru,Business,21499
32,FL0033,Akasa Air,Ahmedabad,Bengaluru,Business,17625
176,FL0177,Vistara,Chennai,Mumbai,Business,17306
108,FL0109,SpiceJet,Srinagar,Delhi,Business,16390


In [54]:
top_10_highest_delayed_flights = data.sort_values("Delay_Minutes", ascending=False).head(10)
top_10_highest_delayed_flights

,Flight_ID,Flight_Date,Airline,Origin,Destination,Aircraft,Travel_Class,Passengers,Seat_Capacity,Average_Ticket_Price,Delay_Minutes,Flight_Status,Weather,Booking_Channel,Avg_Baggage_Kg,Meal_Preference,Passenger_Satisfaction,Load_Factor,Month
50,FL0051,2026-05-03,Air India,Delhi,Mumbai,Airbus A321,Economy,135,210,5129,110.0,Delayed,Rain,Online Travel Portal,17,Vegetarian,4,64.285714,5
26,FL0027,2026-02-10,SpiceJet,Bengaluru,Hyderabad,Airbus A319,Economy,169,180,3973,110.0,Delayed,Clear,Mobile App,7,Vegan,4,93.888889,2
82,FL0083,2026-04-18,Air India Express,Lucknow,Delhi,Airbus A319,Economy,71,180,2484,110.0,Delayed,Cloudy,Online Travel Portal,22,Vegetarian,4,39.444444,4
57,FL0058,2026-04-14,IndiGo,Hyderabad,Kolkata,Boeing 737,Premium Economy,151,210,6834,110.0,Delayed,Cloudy,Travel Agency,16,Vegetarian,4,71.904762,4
177,FL0178,2026-04-30,IndiGo,Delhi,Mumbai,Airbus A319,Economy,112,186,4490,110.0,Delayed,Fog,Online Travel Portal,16,Vegetarian,3,60.215054,4
128,FL0129,2026-01-04,SpiceJet,Lucknow,Delhi,Boeing 737,Premium Economy,67,210,5657,110.0,Delayed,Cloudy,Online Travel Portal,21,Non-Vegetarian,2,31.904762,1
169,FL0170,2026-06-25,IndiGo,Mumbai,Kolkata,Airbus A319,Economy,130,180,3938,110.0,Delayed,Rain,Online Travel Portal,17,No Meal,4,72.222222,6
161,FL0162,2026-01-18,Air India,Hyderabad,Kolkata,Airbus A321,Economy,128,189,5263,110.0,Delayed,Rain,Mobile App,6,Non-Vegetarian,3,67.724868,1
146,FL0147,2026-02-18,IndiGo,Mumbai,Kolkata,Airbus A319,Premium Economy,115,210,9106,110.0,Delayed,Rain,Online Travel Portal,19,Vegetarian,5,54.761905,2
21,FL0022,2026-02-01,Vistara,Kochi,Delhi,Airbus A321,Economy,126,220,5887,75.0,Delayed,Clear,Airline Website,10,Vegetarian,2,57.272727,2


## 5. Grouping & Aggregation

In [55]:
main_summary = data.groupby(
    ["Origin", "Destination"]
).agg(
    Number_of_Flights=("Flight_ID", "count"),
    Total_Passengers=("Passengers", "sum"),
    Average_Delay=("Delay_Minutes", "mean"),
    Average_Ticket_Price=("Average_Ticket_Price", "mean"),
    Average_Satisfaction=("Passenger_Satisfaction", "mean")
).round(2)

main_summary

Number_of_Flights  Total_Passengers  Average_Delay  \
Origin    Destination                                                       
Ahmedabad Bengaluru                   12              1411          10.92   
Bengaluru Hyderabad                   13              1795          15.38   
          Kochi                        8               930           3.12   
Chennai   Mumbai                       9               738           7.44   
Delhi     Chennai                     17              2011          18.18   
          Mumbai                      13              1609          33.92   
          Srinagar                    13              1828          17.85   
Hyderabad Kolkata                      7               884          51.71   
Jaipur    Mumbai                       8               962          10.00   
Kochi     Delhi                       14              1657          14.29   
Lucknow   Delhi                       11              1137          40.00   
Mumbai    Bengaluru                   16              1933          15.44   
          Kolkata                     13              1683          27.00   
Pune      Delhi                       11              1564           5.55   
Srinagar  Delhi                       15              1957          15.60   

                       Average_Ticket_Price  Average_Satisfaction  
Origin    Destination                                              
Ahmedabad Bengaluru                 7857.17                  3.92  
Bengaluru Hyderabad                 4963.15                  3.77  
          Kochi                     5103.12                  3.38  
Chennai   Mumbai                    6713.44                  3.00  
Delhi     Chennai                   7114.82                  3.47  
          Mumbai                    4732.08                  3.54  
          Srinagar                  5589.85                  3.31  
Hyderabad Kolkata                   5399.71                  3.86  
Jaipur    Mumbai                    6044.75                  3.38  
Kochi     Delhi                     4961.14                  3.50  
Lucknow   Delhi                     5318.00                  3.18  
Mumbai    Bengaluru                 6700.81                  3.25  
          Kolkata                   6087.31                  3.92  
Pune      Delhi                     7208.73                  3.73  
Srinagar  Delhi                     7010.53                  3.27

In [56]:
weather_summary = data.groupby("Weather").agg(
    Number_of_Flights=("Flight_ID", "count"),
    Average_Delay=("Delay_Minutes", "mean"),
    Average_Satisfaction=("Passenger_Satisfaction", "mean")
).round(2)

weather_summary

,Number_of_Flights,Average_Delay,Average_Satisfaction
Weather,,,
Clear,34,18.03,3.35
Cloudy,38,18.84,3.84
Fog,35,15.49,3.69
Rain,43,21.42,3.37
Storm,30,19.60,3.17


## 6. Answering Meaningful Questions & Key Findings

In [57]:
ticketprice_economy = data["Average_Ticket_Price"].value_counts()
ticketprice_economy.head(10)

Average_Ticket_Price
3471     2
4468     1
6894     1
3791     1
6505     1
5636     1
9430     1
14173    1
5893     1
2500     1
Name: count, dtype: int64

In [62]:
wether_values = data["Weather"].value_counts()
wether_values

Weather
Rain      43
Cloudy    38
Fog       35
Clear     34
Storm     30
Name: count, dtype: int64

In [63]:
meal_prefrence = data["Meal_Preference"].value_counts()
meal_prefrence

Meal_Preference
Vegetarian        52
Non-Vegetarian    45
Vegan             43
No Meal           40
Name: count, dtype: int64

In [58]:
average_price = data["Average_Ticket_Price"].mean().astype(int)
average_price

np.int64(6111)

In [59]:
maximum_price_ticket = data["Average_Ticket_Price"].max()
maximum_price_ticket

np.int64(27832)

In [60]:
minimum_price_ticket = data["Average_Ticket_Price"].min()
minimum_price_ticket

np.int64(1949)

In [61]:
top_10_highest_ticket_price = data.sort_values("Average_Ticket_Price", ascending=False).head(10)
top_10_highest_ticket_price

,Flight_ID,Flight_Date,Airline,Origin,Destination,Aircraft,Travel_Class,Passengers,Seat_Capacity,Average_Ticket_Price,Delay_Minutes,Flight_Status,Weather,Booking_Channel,Avg_Baggage_Kg,Meal_Preference,Passenger_Satisfaction,Load_Factor,Month
138,FL0139,2026-01-27,Air India Express,Delhi,Chennai,Airbus A321,Business,169,189,27832,20.0,Delayed,Clear,Airline Website,10,Vegan,4,89.417989,1
52,FL0053,2026-05-16,Akasa Air,Mumbai,Bengaluru,Airbus A319,Business,72,210,21499,35.0,Delayed,Clear,Mobile App,15,Non-Vegetarian,2,34.285714,5
32,FL0033,2026-04-27,Akasa Air,Ahmedabad,Bengaluru,Boeing 737,Business,66,180,17625,35.0,Delayed,Cloudy,Mobile App,15,No Meal,4,36.666667,4
176,FL0177,2026-02-17,Vistara,Chennai,Mumbai,Airbus A320,Business,124,180,17306,20.0,Delayed,Cloudy,Mobile App,8,No Meal,5,68.888889,2
108,FL0109,2026-04-12,SpiceJet,Srinagar,Delhi,Airbus A321,Business,129,180,16390,0.0,On Time,Rain,Online Travel Portal,11,Vegan,2,71.666667,4
132,FL0133,2026-02-10,IndiGo,Pune,Delhi,Airbus A321,Business,164,180,15324,0.0,On Time,Storm,Travel Agency,20,Vegetarian,4,91.111111,2
155,FL0156,2026-03-20,Vistara,Mumbai,Bengaluru,Boeing 737,Business,80,186,15040,0.0,On Time,Storm,Online Travel Portal,20,Vegetarian,3,43.010753,3
7,FL0008,2026-04-28,Air India,Delhi,Chennai,Airbus A319,Business,157,220,14173,0.0,On Time,Fog,Airline Website,25,Vegan,5,71.363636,4
51,FL0052,2026-03-07,SpiceJet,Jaipur,Mumbai,Airbus A320,Business,163,210,13824,0.0,On Time,Rain,Travel Agency,6,Vegetarian,2,77.619048,3
92,FL0093,2026-04-30,Air India Express,Srinagar,Delhi,Airbus A319,Business,209,220,12970,0.0,On Time,Clear,Travel Agency,8,Vegan,2,95.000000,4


## 7. Observations & Key Findings

1> Dataset Size: Total 180 flight records and 17 columns are there in the dataset. It contains flight, airline, passenger, ticket price, delay, weather and satisfaction details.

2> Missing Data: Only Delay_Minutes had missing values. There were 7 missing values, which I filled using the median delay. 

3> Ticket Price Scene: Average ticket price is around ₹6,112. The cheapest ticket was ₹1,949, while the most expensive was ₹27,832. Business class tickets were the most expensive.  

4> Delay Problem: Average flight delay is around 19 minutes. 18 flights had delays above 50 minutes, and the highest delay was 110 minutes.

5> Airline Scene: Vistara had the most flights with 37 records. IndiGo and SpiceJet had 32 flights each. IndiGo had the highest average delay at around 25 minutes.

6> Route King: Delhi to Chennai was the busiest route with 17 flights. Hyderabad to Kolkata had the highest average delay, around 51.7 minutes.

7> Weather Scene: Rain had the highest average delay at around 21.4 minutes. Cloudy weather had the highest satisfaction score (3.84), while Storm had the lowest (3.17).

8> Load Factor: Average load factor was around 62.7%. This means flights were carrying around 63% of their total seat capacity on average.

9> Meal Choice: Vegetarian meal was the most preferred choice, with 52 passengers selecting it.

10> Weather King: Rainy weather was observed most often, appearing 43 times in the dataset.